src/Middleware/HeaderPropagation/src/DependencyInjection/HeaderPropagationExtensions.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Http;
+using Microsoft.Extensions.Http.HeaderPropagation;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class HeaderPropagationExtensions
+    {
+        public static IServiceCollection AddHeaderPropagation(this IServiceCollection services, Action<HeaderPropagationOptions> configure)
+        {
+            services.AddHttpContextAccessor();
+            services.Configure(configure);
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IHttpMessageHandlerBuilderFilter, HeaderPropagationMessageHandlerBuilderFilter>());
```

Comment:
I'm not certain that we should plug this into all places in HttpClient factory by default. I think it would be reasonable to take out the handler builder filter and then let users plug in the handler themselves.

The problems I see with the current approach:
- it's really hard to take out something that's "on by default" for one http client configuration
- this approach doesn't let the app developer configure the ordering of message handlers, which can be important

Reply:
Yes agree! the default registration is so confusing that was also throwing me off in the tests as it was also registering on the test client 😅